cpp/modmesh/buffer/pymod/array_common.hpp

```diff
@@ -0,0 +1,307 @@
+#pragma once
+
+/*
+ * Copyright (c) 2024, An-Chi Liu <phy.tiger@gmail.com>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * - Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ * - Neither the name of the copyright holder nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <pybind11/pybind11.h> // Must be the first include.
+
+#include <modmesh/buffer/SimpleArray.hpp>
+#include <modmesh/buffer/pymod/TypeBroadcast.hpp>
+
+namespace modmesh
+{
+
+namespace python
+{
+using slice_type = small_vector<int>;
+using shape_type = modmesh::detail::shape_type;
```

Comment:
`shape_type` should not be in `modmesh::python` namespace as it has been in `modmesh::detail` namespace.

Reply:
maybe we can also put it `modmesh::detail`. I couldn't find a good place for it. Maybe `SimpleArray.hpp` is fine, but it might be better to have a header file for basic types.

Reply:
`shape_type` is still in `modmesh::python` namespace.  This should be fixed.  Please remove this alias When using `shape_type`, spelling out the full scope `modmesh::detail::shape_type`.

Perhaps `slice_type` should be treated in the same way.